File: backend/app/schemas/ai.py
"""Pydantic v2 schemas for AI chat operations."""

import uuid
from typing import Optional, List

from pydantic import BaseModel


class ChatRequest(BaseModel):
    """AI chat request schema."""
    message: str
    session_id: Optional[uuid.UUID] = None
    language: str = "en"


class ChatResponse(BaseModel):
    """AI chat response schema."""
    response: str
    session_id: uuid.UUID


class ChatMessage(BaseModel):
    """Individual chat message for history."""
    role: str  # "user" or "assistant"
    content: str
    timestamp: str


class ChatHistory(BaseModel):
    """Chat history response."""
    session_id: uuid.UUID
    messages: List[ChatMessage]

File: frontend/src/services/ai.ts
import { api } from "./api";

export const aiApi = {
  chat: async (message: string, sessionId?: string, language?: string): Promise<{ response: string; session_id: string }> => {
    const { data } = await api.post("/ai/chat", { message, session_id: sessionId, language });
    return data;
  },

  getHistory: async (sessionId: string): Promise<{ messages: Array<{ role: string; content: string; timestamp: string }> }> => {
    const { data } = await api.get(`/ai/chat/history/${sessionId}`);
    return data;
  },

  clearHistory: async (sessionId: string): Promise<void> => {
    await api.delete(`/ai/chat/history/${sessionId}`);
  },
};
